src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -742,5 +748,103 @@ private void UpdateCacheExpirationTime(KeyValueWatcher changeWatcher, bool succe
 
             changeWatcher.CacheExpires = DateTimeOffset.UtcNow.Add(cacheExpirationTime);
         }
+
+        private async Task ExecuteWithFailOverPolicyAsync(Func<ConfigurationClient, Task> funcToExecute, CancellationToken cancellationToken = default)
+        {
+            using IEnumerator<ConfigurationClient> clientEnumerator = _configClientManager.GetAvailableClients().GetEnumerator();
+
+            // Guaranteed to return at least 1 element
+            Debug.Assert(clientEnumerator.MoveNext());

```

Comment:
Does this mean we won't call `clientEnumerator.MoveNext()` in release build?

Reply:
Updated the logic, we call `clientEnumerator.MoveNext()` every time.